legend-engine-pure-code-compiled-core/src/main/resources/core/pure/graphFetch/tests/testPropertyTreeConversion.pure

```diff
@@ -325,3 +325,870 @@ Class meta::pure::graphFetch::tests::propertyTreeConversion::Z
+                    // case 2 :  owner class is same as class in property path tree      [ownerclass :Street]    >>  GFT Expected:    {Street(street)}
+
+            //  case c : access on basetype + single subtype properts                                                       : root{ c_Location{ p_Location.coordinates}, c_Street {p_Street.coordinates,  p_Location.street}}  
+                    // case 1 :  owner class is superType of class in property path tree [ownerclass :Location]  >>  GFT Expected:    {Location(coordinates, ->subtype(Street)(coordinates, street))}
+                    // case 2 :  owner class is same as class in property path tree      [ownerclass :Street]    >>  GFT Expected:    {Street(coordinates,  street)}
+
+
+       // case D: property tree has properties accessed from supertype and multiple subtypes
+
+             //  case a : access on basetype properts only                                                                    :     root{  c_Location{ p_Location.coordinates}, c_Street{ p_Street.coordinates}, c_Road{ p_Road.coordinates}}}   
+                    // case 1 :  owner class is superType of class in property path tree [ownerclass :Location]  >>  GFT Expected:    {Location(coordinates, ->subtype(Street)(coordinates),  ->subtype(Road)(coordinates)}
+                    // case 2 :  owner class is same as class in property path tree      [ownerclass :Street]    >>  GFT Expected:    {Street(coordinates)}
+
+            //  case b : access on subtype properts only                                                                     :   root{  c_Location{ p_Location.coordinates}, c_Street{ p_Street.street}, c_Road{ p_Road.road}}  
+                    // case 1 :  owner class is superType of class in property path tree [ownerclass :Location]  >>  GFT Expected:    {Location(coordinates, ->subtype(Street)(street, coordinates), ->subtype(Road, coordinates)(road))}
+                    // case 2 :  owner class is same as class in property path tree      [ownerclass :Street]    >>  GFT Expected:    {Street(street)}
+
+            //  case c : access on basetype + single subtype properts                                                       : root{ c_Location{ p_Location.coordinates}, c_Street {p_Street.coordinates,  p_Location.street},  c_Road{ p_Road_coordinates, p_Road.road}}  
+                    // case 1 :  owner class is superType of class in property path tree [ownerclass :Location]  >>  GFT Expected:    {Location(coordinates, ->subtype(Street)(coordinates, street),  ->subtype(Road)(coordinates, road) )}
+                    // case 2 :  owner class is same as class in property path tree      [ownerclass :Street]    >>  GFT Expected:    {Street(coordinates,  street)}
+//
+
+
+function <<meta::pure::profiles::test.Test>> meta::pure::graphFetch::tests::propertyTreeConversion::tesSubTypeOnRoot_accessFromBaseType_baseTypeProperties():Boolean[1]
```

Comment:
All these tests can be made much simpler using some helper functions (and imports)